sdk/ai/azure-ai-projects/samples/agents/user_functions.py

```diff
@@ -30,17 +30,21 @@ def fetch_current_datetime(format: Optional[str] = None) -> str:
     return time_json
 
 
-def fetch_weather(location: str) -> str:
+def fetch_weather(location: str, unit: str) -> str:
     """
     Fetches the weather information for the specified location.
 
     :param location (str): The location to fetch weather for.
+    :param unit (str): The temperature unit to use.
     :return: Weather information as a JSON string.
     :rtype: str
     """
     # In a real-world scenario, you'd integrate with a weather API.
     # Here, we'll mock the response.
     mock_weather_data = {"New York": "Sunny, 25°C", "London": "Cloudy, 18°C", "Tokyo": "Rainy, 22°C"}
+    if unit == "Fahrenheit":
```

Comment:
Should this be a case-insensitive check?

Reply:
I actually use case sensitive check on purpose.   Because in the samples, I apply enum with the word, 'Celsius' and 'Fahrenheit'.   I want to give the impression here that the exact word will be used.